extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/JaegerPropagator.java

```diff
@@ -108,16 +118,27 @@ public static JaegerPropagator getInstance() {
     setter.set(carrier, PROPAGATION_HEADER, new String(chars));
   }
 
+  private static <C> void injectBaggage(Baggage baggage, C carrier, Setter<C> setter) {
+    for (Entry entry : baggage.getEntries()) {
+      setter.set(carrier, BAGGAGE_PREFIX + entry.getKey(), entry.getValue());
+    }
+  }
+
   @Override
   public <C> Context extract(Context context, @Nullable C carrier, Getter<C> getter) {
     Objects.requireNonNull(getter, "getter");
 
     SpanContext spanContext = getSpanContextFromHeader(carrier, getter);
-    if (!spanContext.isValid()) {
-      return context;
+    if (spanContext.isValid()) {
+      context = context.with(Span.wrap(spanContext));
+    }
+
+    Baggage baggage = getBaggageFromHeader(carrier, getter);
+    if (baggage != null) {
```

Comment:
Does it make sense to add empty baggage similar to how we add invalid span?

Reply:
That's a tricky one (I will let @pavolloffay @jpkrohling comment), but I'd imagine we wanting to keep whatever `Baggage` we had before, in case there was no baggage *at all*.